course_contents/2_intro_to_python/notes/03. Comprehensions.md

```diff
@@ -0,0 +1,99 @@
+### The list comprehension
+- the list comprehension allows us to create a list without manually appending objects or running loops
+- the general expression for a list comprehension is:
+
+```py
+new_collection = [element for element in collection] 
+```
+
+- the `element` variable will take the value of each element from the `collection` variable and be appended to the `new_collection`
+
+An example of a list comprehension:
+
+```py
+numbers = [i for i in range(5)]
+print(numbers)
+```
+
+#### List comprehension with conditionals
+- conditionals can be added to list comprehensions to make them more versatile
+
+An example of a list comprehension with a conditional:
+
+```py
+numbers = [i for i in range(5) if i % 2 == 0] #appending even numbers only
+print(numbers)
+```
+
+### Set comprehension
+- set comprehensions follow the same syntax as a list comprehension
+- the only difference is that curly brackets are used instead of square brackets
+
+An example of a set comprehension:
+
+```py
+collection = [1, 2, 2, 3, 4, 3, 5]
+numbers = {i for i in collection}
+print(numbers)
+```
+
+### Dictionary comprehension
+- dictionary comprehensions follow the next syntax:
+
+```py
+dictionary = {
+    collection_1[i] : collection_2[i]
+    for i in collection 
+}
+```
+
+- the collection_1[i] will be the ith element from the collection_1 variable and it will be the key
+- the collection_2[i] will be the ith element from the collection_2 variable and it will be the value
+
+An example of a dictionary comprehension:
+
+```py
+friends = ["Rolf", "Dane", "Josh"]
+numbers = [1,2,3]
+dictionary = {
+    friends[i] : numbers[i]
+    for i in range(3)
+}
+print(dictionary)
+```
+
+### Comprehensions with conditions
```

Comment:
You've already touched on this on line 18

Reply:
^ Yep, I think we can just delete lines 75 to 90?